gtsam_unstable/slam/PartialPriorFactor.h

```diff
@@ -11,142 +11,137 @@
+        prior_(Vector1(prior)),
+        indices_(1, idx) {
+    assert(model->dim() == 1);
+  }
+
+  /** Indices Constructor: Specify the relevant measured indices in the parameter vector.*/
+  PartialPriorFactor(Key key, const std::vector<size_t>& indices, const Vector& prior,
+                     const SharedNoiseModel& model)
+      : Base(model, key),
         prior_(prior),
         indices_(indices) {
-      assert((size_t)prior_.size() == indices_.size());
-      assert(model->dim() == (size_t)prior.size());
-    }
+    assert((size_t)prior_.size() == indices_.size());
+    assert((size_t)prior.size() == model->dim());
+  }
+
+  /** Implement functions needed for Testable */
+
+  /** print */
+  void print(const std::string& s, const KeyFormatter& keyFormatter = DefaultKeyFormatter) const override {
+    Base::print(s, keyFormatter);
+    gtsam::print(prior_, "Prior: ");
+    std::cout << "Indices: ";
+    for (const int i : indices_) { std::cout << i << " " ;}
+    std::cout << std::endl;
+  }
+
+  /** equals */
+  bool equals(const NonlinearFactor& expected, double tol=1e-9) const override {
+    const This *e = dynamic_cast<const This*> (&expected);
+    return e != nullptr && Base::equals(*e, tol) &&
+        gtsam::equal_with_abs_tol(this->prior_, e->prior_, tol) &&
+        this->indices_ == e->indices_;
+  }
 
-    /// @return a deep copy of this factor
-    gtsam::NonlinearFactor::shared_ptr clone() const override {
-      return boost::static_pointer_cast<gtsam::NonlinearFactor>(
-          gtsam::NonlinearFactor::shared_ptr(new This(*this))); }
+  /**
+   * Evaluate the error h(x) - prior, where h(x) returns a parameter vector
+   * representation of x that is consistent with the prior. Note that this
+   * function expects any derived factor to have overridden the Parameterize()
+   * member function.
+   */
+  Vector evaluateError(const T& x, boost::optional<Matrix&> H = boost::none) const override {
+    // Extract the relevant subset of the parameter vector and Jacobian.
+    Matrix H_full;
+    const Vector hx_full = Parameterize(x, H ? &H_full : nullptr);
 
-    /** implement functions needed for Testable */
+    Vector hx(indices_.size());
+    if (H) (*H) = Matrix::Zero(indices_.size(), T::dimension);
 
-    /** print */
-    void print(const std::string& s, const KeyFormatter& keyFormatter = DefaultKeyFormatter) const override {
-      Base::print(s, keyFormatter);
-      gtsam::print(prior_, "prior");
+    for (size_t i = 0; i < indices_.size(); ++i) {
+      hx(i) = hx_full(indices_.at(i));
+      if (H) (*H).row(i) = H_full.row(indices_.at(i));
     }
 
-    /** equals */
-    bool equals(const NonlinearFactor& expected, double tol=1e-9) const override {
-      const This *e = dynamic_cast<const This*> (&expected);
-      return e != nullptr && Base::equals(*e, tol) &&
-          gtsam::equal_with_abs_tol(this->prior_, e->prior_, tol) &&
-          this->indices_ == e->indices_;
-    }
+    return hx - prior_;
```

Comment:
@miloknowles you may want to use the manifold version here which is `return -traits<T>::Local(x, prior_);`